src/serialize/value.rs

```diff
@@ -0,0 +1,1075 @@
+                            typ,
+                            BuiltinTypeCheckErrorKind::MismatchedType { expected },
+                        )));
+                    }
+                };
+                tuple.serialize(typ, cell_writer)
+            }
+            _ => Err(SerializationError::new(
+                PythonDriverSerializationError::UnknownColumnType,
+            )),
+        }
+    }
+}
+
+impl<'a, 'py> SerializeValue for PyAnyWrapper<'a, 'py> {
+    fn serialize<'b>(
+        &self,
+        typ: &ColumnType,
+        cell_writer: CellWriter<'b>,
+    ) -> Result<WrittenCellProof<'b>, SerializationError> {
+        if self.is_none() {
+            return Ok(cell_writer.set_null());
+        }
+
+        self.serialize_types(typ, cell_writer)
+    }
+}
+
+#[derive(Debug)]
+pub(crate) struct PyListWrapper<'a, 'py> {
+    pub(crate) inner: &'a Bound<'py, PyList>,
+}
+
+impl<'a, 'py> Deref for PyListWrapper<'a, 'py> {
+    type Target = Bound<'py, PyList>;
+
+    fn deref(&self) -> &Self::Target {
+        self.inner
+    }
+}
+
+impl<'a, 'py> PyListWrapper<'a, 'py> {
+    fn new(value: &PyAnyWrapper<'a, 'py>) -> PyResult<Self> {
+        let list: &Bound<PyList> = value.downcast::<PyList>()?;
+        Ok(PyListWrapper {
+            inner: list,
+            // inner: Bound::clone(list),
+        })
+    }
+}
+
+impl<'a, 'py> SerializeValue for PyListWrapper<'a, 'py> {
+    fn serialize<'b>(
+        &self,
+        typ: &ColumnType,
+        cell_writer: CellWriter<'b>,
+    ) -> Result<WrittenCellProof<'b>, SerializationError> {
+        let items = self.iter();
+
+        serialize_sequence(
+            std::any::type_name::<PyList>(),
+            items.len(),
+            items,
+            typ,
+            cell_writer,
+        )
+    }
+}
+
+pub(crate) struct PySetWrapper<'py, 'a>(&'a Bound<'py, PySet>);
+
+impl<'a, 'py> Deref for PySetWrapper<'a, 'py> {
+    type Target = Bound<'py, PySet>;
+
+    fn deref(&self) -> &Self::Target {
+        self.0
+    }
+}
+
+impl<'a, 'py> PySetWrapper<'a, 'py> {
+    fn new(value: &PyAnyWrapper<'a, 'py>) -> PyResult<Self> {
+        let set: &Bound<PySet> = value.downcast::<PySet>()?;
+        Ok(PySetWrapper(set))
+    }
+}
+
+impl<'a, 'py> SerializeValue for PySetWrapper<'a, 'py> {
+    fn serialize<'b>(
+        &self,
+        typ: &ColumnType,
+        cell_writer: CellWriter<'b>,
+    ) -> Result<WrittenCellProof<'b>, SerializationError> {
+        let items = self.iter();
+
+        serialize_sequence(
+            std::any::type_name::<PySet>(),
+            items.len(),
+            items,
+            typ,
+            cell_writer,
+        )
+    }
+}
+
+fn serialize_sequence<'t, 'b, 'py>(
+    rust_name: &'static str,
+    len: usize,
+    iter: impl Iterator<Item = Bound<'py, PyAny>>,
+    typ: &ColumnType,
+    writer: CellWriter<'b>,
+) -> Result<WrittenCellProof<'b>, SerializationError> {
```

Comment:
:wrench: This can also be generic over `T: Any` instead of accepting `rust_name`.